commit 732e099ad8e994345381a2b41ec2e26a7cf2fce2
Author: sh!zeeg <shizeeque@gmail.com>
Date:   Wed Jun 26 20:13:51 2013 +0400

    add "window-title-curpage" option to display [current page / total pages]

diff --git a/config.c b/config.c
index 1be86d8..7003d2e 100644
--- a/config.c
+++ b/config.c
@@ -207,6 +207,8 @@ config_load_default(zathura_t* zathura)
   bool_value = true;
   girara_setting_add(gsession, "abort-clear-search",     &bool_value,  BOOLEAN, false, _("Clear search results on abort"), NULL, NULL);
   bool_value = false;
+  girara_setting_add(gsession, "window-title-curpage",   &bool_value,  BOOLEAN, false, _("Show current page / page total in window title"), NULL, NULL);
+  bool_value = false;
   girara_setting_add(gsession, "window-title-basename",  &bool_value,  BOOLEAN, false, _("Use basename of the file in the window title"), NULL, NULL);
   bool_value = false;
   girara_setting_add(gsession, "statusbar-basename",     &bool_value,  BOOLEAN, false, _("Use basename of the file in the statusbar"), NULL, NULL);
diff --git a/zathura.c b/zathura.c
index cdf4eac..424aa43 100644
--- a/zathura.c
+++ b/zathura.c
@@ -648,6 +648,9 @@ document_open(zathura_t* zathura, const char* path, const char* password,
     zathura_document_set_adjust_mode(document, ZATHURA_ADJUST_NONE);
   }
 
+  /* update title */
+  window_title_update(zathura);
+
   /* update statusbar */
   bool basename_only = false;
   girara_setting_get(zathura->ui.session, "statusbar-basename", &basename_only);
@@ -765,17 +768,6 @@ document_open(zathura_t* zathura, const char* path, const char* password,
   /* bookmarks */
   zathura_bookmarks_load(zathura, file_path);
 
-  /* update title */
-  basename_only = false;
-  girara_setting_get(zathura->ui.session, "window-title-basename", &basename_only);
-  if (basename_only == false) {
-    girara_set_window_title(zathura->ui.session, file_path);
-  } else {
-    char* tmp = g_path_get_basename(file_path);
-    girara_set_window_title(zathura->ui.session, tmp);
-    g_free(tmp);
-  }
-
   g_free(file_uri);
 
   /* adjust window */
@@ -1022,6 +1014,38 @@ error_out:
 }
 
 void
+window_title_update(zathura_t* zathura)
+{
+  if (zathura == NULL) {
+    return;
+  }
+
+  if (zathura->document == NULL) {
+    girara_set_window_title(zathura->ui.session, _("[No name]"));
+    return;
+  }
+  unsigned int number_of_pages     = zathura_document_get_number_of_pages(zathura->document);
+  unsigned int current_page_number = zathura_document_get_current_page_number(zathura->document);
+  const char*  file_path           = zathura_document_get_path(zathura->document);
+
+  bool show_current_page_number    = false;
+  bool basename_only               = false;
+  girara_setting_get(zathura->ui.session, "window-title-curpage", &show_current_page_number);
+  girara_setting_get(zathura->ui.session, "window-title-basename", &basename_only);
+
+  char* tmp = g_path_get_basename(file_path);
+
+  if (show_current_page_number == false) {
+    girara_set_window_title(zathura->ui.session, (basename_only) ? tmp : file_path);
+  }  else {
+    char* title = g_strdup_printf("[%d/%d]: %s", current_page_number + 1, number_of_pages, (basename_only) ? tmp : file_path);
+    girara_set_window_title(zathura->ui.session, title);
+    g_free(title);
+  }
+  g_free(tmp);
+}
+
+void
 statusbar_page_number_update(zathura_t* zathura)
 {
   if (zathura == NULL || zathura->ui.statusbar.page_number == NULL) {
@@ -1038,6 +1062,8 @@ statusbar_page_number_update(zathura_t* zathura)
   } else {
     girara_statusbar_item_set_text(zathura->ui.session, zathura->ui.statusbar.page_number, "");
   }
+  /* update pages in window title also, if window-title-curpage is set */
+  window_title_update(zathura);
 }
 
 void
diff --git a/zathura.h b/zathura.h
index fd1d626..e9aa3ca 100644
--- a/zathura.h
+++ b/zathura.h
@@ -335,6 +335,13 @@ void page_widget_set_mode(zathura_t* zathura, unsigned int pages_per_row, unsign
 void statusbar_page_number_update(zathura_t* zathura);
 
 /**
+ * Updates window title
+ *
+ * @param zathura The zathura session
+ */
+void window_title_update(zathura_t* zathura);
+
+/**
  * Checks whether current jump has a previous jump
  *
  * @param zathura The zathura session
